integrations/linear/src/index.ts

```diff
@@ -0,0 +1,423 @@
+    return createTrigger(this.source, events.onIssueUpdated, params);
+  }
+
+  onIssueLabel(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onIssueLabel, params);
+  }
+
+  onIssueLabelCreated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onIssueLabelCreated, params);
+  }
+
+  onIssueLabelRemoved(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onIssueLabelRemoved, params);
+  }
+
+  onIssueLabelUpdated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onIssueLabelUpdated, params);
+  }
+
+  onIssueSLA(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onIssueSLA, params);
+  }
+
+  onProject(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProject, params);
+  }
+
+  onProjectCreated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectCreated, params);
+  }
+
+  onProjectRemoved(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectRemoved, params);
+  }
+
+  onProjectUpdated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectUpdated, params);
+  }
+
+  onProjectUpdate(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectUpdate, params);
+  }
+
+  onProjectUpdateCreated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectUpdateCreated, params);
+  }
+
+  onProjectUpdateRemoved(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectUpdateRemoved, params);
+  }
+
+  onProjectUpdateUpdated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onProjectUpdateUpdated, params);
+  }
+
+  onReaction(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onReaction, params);
+  }
+
+  onReactionCreated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onReactionCreated, params);
+  }
+
+  onReactionRemoved(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onReactionRemoved, params);
+  }
+
+  onReactionUpdated(params: TriggerParams = {}) {
+    return createTrigger(this.source, events.onReactionUpdated, params);
+  }
+
+  webhooks() {
+    return new Webhooks(this.runTask.bind(this));
+  }
+}
+
+export function onError(error: unknown) {
+  if (!(error instanceof RatelimitedLinearError)) {
+    return;
+  }
+
+  const rateLimitRemaining = error.raw?.response?.headers?.get("X-RateLimit-Requests-Remaining");
+  const rateLimitReset = error.raw?.response?.headers?.get("X-RateLimit-Requests-Reset");
+
+  if (rateLimitRemaining === "0" && rateLimitReset) {
+    const resetDate = new Date(Number(rateLimitReset) * 1000);
+
+    return {
+      retryAt: resetDate,
+      error,
+    };
+  }
+
+  const queryComplexity = error.raw?.response?.headers?.get("X-Complexity");
```

Comment:
This is really cool 😸 I didn't know they did this. I've never created a GraphQL API, I'm guessing this is more standard in them?

Reply:
Indeed. Nesting, batching, circular references (hopefully not), etc. Simple rate limits just don't do it here.